state-sync/state-sync-v2/state-sync-multiplexer/src/lib.rs

```diff
@@ -50,31 +87,67 @@ impl StateSyncMultiplexer {
             Err(error) => panic!("Failed to fetch the initial synced version: {:?}", error),
         }
 
-        // TODO(joshlind): use configs to determine the state sync v1 mode
-        let state_sync_v1_read_only = false;
-
-        // Create a state sync v1 bootstrapper
-        let state_sync_bootstrapper = StateSyncBootstrapper::bootstrap(
-            network,
-            mempool_notifier,
-            consensus_listener,
-            storage,
-            executor,
-            node_config,
-            waypoint,
-            event_subscription_service,
-            state_sync_v1_read_only,
-        );
+        // TODO(joshlind): update this to support also running v1 in read-only mode!
+        // Start state sync (with the version depending on the config)
+        let mut state_sync_v1 = None;
+        let mut state_sync_v2 = None;
+        let activate_state_sync_v2 = node_config
+            .state_sync
+            .state_sync_driver
+            .activate_state_sync_v2;
+        if activate_state_sync_v2 {
+            // Start the state sync v2 driver
+            state_sync_v2 = Some(DriverFactory::create_and_spawn_driver(
+                true,
+                node_config,
+                waypoint,
+                storage,
+                executor,
+                mempool_notifier,
+                consensus_listener,
+                event_subscription_service,
+                streaming_service_client,
+            ));
+        } else {
+            // Start state sync v1
+            state_sync_v1 = Some(StateSyncBootstrapper::bootstrap(
+                network,
+                mempool_notifier,
+                consensus_listener,
+                storage.reader,
+                executor,
+                node_config,
+                waypoint,
+                event_subscription_service,
+                false,
+            ));
+        }
```

Comment:
Yeah, there's a TODO(...) that I added above to do this (see line 90). Unfortunately, it's not as simple as doing read_only to true because I need to refactor state sync v1 to take in different objects when in read_only mode.